### ISSUE TITLE
Composer dependency fail for PHP mongodb driver/extension

### ISSUE THREAD

**Issue by philicious** [dreamfactorysoftware/df-docker#19]
As of recently (since DF 2.1.2), when building the image, I get
```
  Problem 1
    - dreamfactory/df-mongodb 0.2.3 requires jenssegers/mongodb 3.0.* -> satisfiable by jenssegers/mongodb[v3.0.2, v3.0.0, v3.0.0-alpha, v3.0.0-beta, v3.0.1].
    - dreamfactory/df-mongodb 0.2.3 requires jenssegers/mongodb 3.0.* -> satisfiable by jenssegers/mongodb[v3.0.2, v3.0.0, v3.0.0-alpha, v3.0.0-beta, v3.0.1].
    - jenssegers/mongodb v3.0.0 requires mongodb/mongodb ^1.0.0 -> satisfiable by mongodb/mongodb[1.0.2, 1.0.0, 1.0.0-alpha1, 1.0.0-beta1, 1.0.0-beta2, 1.0.1, 1.1.x-dev, v1.0.x-dev].
    - jenssegers/mongodb v3.0.0-alpha requires mongodb/mongodb ^1.0.0 -> satisfiable by mongodb/mongodb[1.0.2, 1.0.0, 1.0.0-alpha1, 1.0.0-beta1, 1.0.0-beta2, 1.0.1, 1.1.x-dev, v1.0.x-dev].
    - jenssegers/mongodb v3.0.0-beta requires mongodb/mongodb ^1.0.0 -> satisfiable by mongodb/mongodb[1.0.2, 1.0.0, 1.0.0-alpha1, 1.0.0-beta1, 1.0.0-beta2, 1.0.1, 1.1.x-dev, v1.0.x-dev].
    - jenssegers/mongodb v3.0.1 requires mongodb/mongodb ^1.0.0 -> satisfiable by mongodb/mongodb[1.0.2, 1.0.0, 1.0.0-alpha1, 1.0.0-beta1, 1.0.0-beta2, 1.0.1, 1.1.x-dev, v1.0.x-dev].
    - jenssegers/mongodb v3.0.2 requires mongodb/mongodb ^1.0.0 -> satisfiable by mongodb/mongodb[1.0.2, 1.0.0, 1.0.0-alpha1, 1.0.0-beta1, 1.0.0-beta2, 1.0.1, 1.1.x-dev, v1.0.x-dev].
    - jenssegers/mongodb v3.0.2 requires mongodb/mongodb ^1.0.0 -> satisfiable by mongodb/mongodb[1.0.2, 1.0.0, 1.0.0-alpha1, 1.0.0-beta1, 1.0.0-beta2, 1.0.1, 1.1.x-dev, v1.0.x-dev].
    - mongodb/mongodb 1.0.2 requires ext-mongodb ^1.1.0 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb v1.0.x-dev requires ext-mongodb ^1.1.0 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb 1.1.x-dev requires ext-mongodb ^1.1.0 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb 1.0.2 requires ext-mongodb ^1.1.0 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb 1.0.1 requires ext-mongodb ^1.1.0 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb 1.0.0-beta2 requires ext-mongodb ^1.1.1 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb 1.0.0-beta1 requires ext-mongodb ^1.0.0 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb 1.0.0-alpha1 requires ext-mongodb ^1.0.0 -> the requested PHP extension mongodb is missing from your system.
    - mongodb/mongodb 1.0.0 requires ext-mongodb ^1.1.0 -> the requested PHP extension mongodb is missing from your system.
```

Now the reason for this is that Ubuntu14's `php5-mongo` has version 1.4.5, which is the legacy driver (https://pecl.php.net/package/mongo) but we need the new driver (https://pecl.php.net/package/mongodb) which isnt available in Ubuntu repos. 

I think this regression bug got introduced with the latest changes to https://github.com/dreamfactorysoftware/df-mongodb (changing the mongodb driver)

I'm working on a fix right now. Unfortunately we got to install php-mongodb extension via PECL now and pull in several system packages for building the extension from source.

**Comment by philicious** [comment]
I openend a request at Ubuntu Launchpad to get the new php-mongodb driver also for trusty-universe (Ubuntu14). Its available in xenial but built against PHP7 API. https://bugs.launchpad.net/ubuntu/+source/php-mongodb/+bug/1579815

If it should become available in trusty-universe, image built time and resulting size can be reduced again.

**Comment by todda48** [comment]
Thanks for that request!

**Comment by philicious** [comment]
fixed and merged with #21
